app/controllers/local_authorities_controller.rb

```diff
@@ -2,4 +2,8 @@ class LocalAuthoritiesController < ApplicationController
   def index
     @authorities = LocalAuthority.all.order(name: :asc)
   end
+
+  def show
+    @authority = LocalAuthority.find_by_slug(params[:slug])
```

Comment:
`find_by_slug` will return `nil` if the provided slug doesn't map to an actual authority so we should prefer using the bang! variant `find_by_slug!` which will raise an `ActiveRecord::RecordNotFound` exception.  By default rails will catch those exceptions and render a 404, whereas returning a `nil` will most likely end up causing a `NoMethodError` exception somewhere and rendering 500.